geograypher/meshes/derived_meshes.py

```diff
@@ -514,3 +514,274 @@ def aggregate_projected_images(
+                If True, enables frustum culling to exclude mesh faces outside the camera's view,
+                Defaults to False.
+
+        Returns:
+            np.ndarray: For each camera, returns an array of face indices corresponding to each pixel
+            in the image. Indices are adjusted for batch offsets and set to -1 where no valid face is
+            found. If the input is a single PhotogrammetryCamera, the shape is (h, w). If it's a camera
+            set, then it is (n_cameras, h, w).
+        """
+
+        # Create a camera from the metashape parameters
+        if isinstance(cameras, PhotogrammetryCamera):
+            p3d_cameras = self.get_single_pytorch3d_camera(
+                device=self.device, camera=cameras
+            )
+            image_size = cameras.get_image_size(image_scale=render_img_scale)
+        else:
+            p3d_cameras = self.transform_into_pytorch3d_camera_set(
+                device=self.device, cameras=cameras
+            )
+            image_size = cameras[0].get_image_size(image_scale=render_img_scale)
+
+        raster_settings = self.RasterizationSettings(
+            image_size=image_size,
+            blur_radius=0.0,
+            faces_per_pixel=1,
+            cull_to_frustum=cull_to_frustum,
+        )
+
+        # Don't wrap this in a MeshRenderer like normal because we need intermediate results
+        rasterizer = self.MeshRasterizer(
+            cameras=p3d_cameras, raster_settings=raster_settings
+        ).to(self.device)
+
+        # Create a pytorch3d mesh
+        pytorch3d_mesh = self.create_pytorch3d_mesh(batch_size=len(p3d_cameras))
+
+        # Perform the expensive pytorch3d operation
+        fragments = rasterizer(pytorch3d_mesh)
+
+        # Extract pix_to_face from fragments, move it to the CPU, and convert tensor to NumPy array
+        pix_to_face = fragments.pix_to_face.cpu().numpy()
+
+        # Removes last dimension which is number of faces that can corrrespond to pixel
+        # pix_to_face now is (batch_size, height, width)
+        pix_to_face = pix_to_face[:, :, :, 0]
+
+        # Create an array mask to note where pix_to_face correspondances are -1 (invalid)
+        invalid_mask = pix_to_face == -1
+
+        # Track batch index offset to account for mesh extension
+        offset_array = np.arange(
+            0,
+            self.pyvista_mesh.n_faces * pix_to_face.shape[0],
+            self.pyvista_mesh.n_faces,
+        )
+
+        # Convert dimensions of offset_array from (batch_size,) to (batch_size, 1, 1) in order to match pix_to_face dimensions
+        offset_array = np.expand_dims(offset_array, (1, 2))
+
+        # Adjust pix_to_face indices based on the batch offset
+        pix_to_face = pix_to_face - offset_array
+
+        # Add -1 (invalid) values back into pix_to_face
+        pix_to_face[invalid_mask] = -1
+
+        return pix_to_face
+
+    def get_single_pytorch3d_camera(self, device: str, camera: PhotogrammetryCamera):
```

Comment:
Since we have `self.device` and you can always change what device an object is on, I'd personally get rid of the `device` argument. Also, I generally like to order the arguments from most to least important. This is mostly a style choice, but it also makes it less distributive if you remove one of the less important arguments or give it a default value.